easybuild/scripts/rpath_args.py

```diff
@@ -0,0 +1,119 @@
+
+# wheter or not to use -Wl to pass options to the linker
+if cmd in ['ld', 'ld.gold']:
+    flag_prefix = ''
+else:
+    flag_prefix = '-Wl,'
+
+if rpath_filter:
+    rpath_filter = rpath_filter.split(',')
+else:
+    # don't RPATH system library locations by default
+    rpath_filter = ['/lib', '/usr/lib']
+
+version_mode = False
+cmd_args, cmd_args_rpath = [], []
+
+# process list of original command line arguments
+idx = 0
+while idx < len(args):
+
+    arg = args[idx]
+
+    # if command is run in 'version check' mode, make sure we don't include *any* -rpath arguments
+    if arg in ['-v', '-V', '--version', '-dumpversion']:
+        version_mode = True
+        cmd_args.append(arg)
+
+    # FIXME: also consider $LIBRARY_PATH?
+    # FIXME: support to hard inject additional library paths?
+    # FIXME: support to specify list of path prefixes that should not be RPATH'ed into account?
+    # FIXME skip paths in /tmp, build dir, etc.?
+
+    # handle -L flags, inject corresponding -rpath flag
+    elif arg.startswith('-L'):
+        # take into account that argument to -L may be separated with one or more spaces...
+        if arg == '-L':
+            # actual library path is next argument when arg='-L'
+            idx += 1
+            lib_path = args[idx]
+        else:
+            lib_path = arg[2:]
+
+        if os.path.isabs(lib_path) and not any(lib_path.startswith(f) for f in rpath_filter):
+            # inject -rpath flag in front for every -L with an absolute path,
+            # also retain the -L flag (without reordering!)
+            cmd_args_rpath.append(flag_prefix + '-rpath=%s' % lib_path)
+            cmd_args.append('-L%s' % lib_path)
+        else:
+            # don't RPATH in relative paths;
+            # it doesn't make much sense, and it can also break the build because it may result in reordering lib paths
+            cmd_args.append('-L%s' % lib_path)
+
+    # filter out --enable-new-dtags if it's used;
+    # this would result in copying rpath to runpath, meaning that $LD_LIBRARY_PATH is taken into account again
+    elif arg != '--enable-new-dtags':
+        cmd_args.append(arg)
+
+    idx += 1
+
+# add -rpath flags in front
+cmd_args = cmd_args_rpath + cmd_args
+
+if not version_mode:
+    cmd_args = [
+        # always include '$ORIGIN/../lib' and '$ORIGIN/../lib64'
+        # $ORIGIN will be resolved by the loader to be the full path to the 'executable'
+        # see also https://linux.die.net/man/8/ld-linux;
+        flag_prefix + '-rpath=$ORIGIN/../lib',
+        flag_prefix + '-rpath=$ORIGIN/../lib64',
+        # try to make sure that RUNPATH is not used by always injecting --disable-new-dtags
+        flag_prefix + '--disable-new-dtags',
+    ] + cmd_args
+
+# wrap all arguments into single quotes to avoid further bash expansion
+cmd_args = ["'%s'" % arg.replace("'", "''") for arg in cmd_args]
+
+# output: statement to define $CMD_ARGS and $RPATH_ARGS
+print "CMD_ARGS=(%s)" % ' '.join(cmd_args)
```

Comment:
Doesn't matter much since most of the rest of EB isn't Py3 compatible yet, but I guess it doesn't hurt looking forward, I'll change this, thanks.